src/test/java/org/zalando/fahrschein/CursorManagerTest.java

```diff
@@ -0,0 +1,78 @@
+package org.zalando.fahrschein;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.zalando.fahrschein.domain.Cursor;
+import org.zalando.fahrschein.domain.Partition;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class CursorManagerTest {
+
+    /*
+    private static Object[] parameter(String name, @Nullable String initialOffset, String oldestAvailableOffset, String newestAvailableOffset, @Nullable String expectedOffset) {
+
+    }
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> parameters() {
+
+        return asList(parameter(""))
+
+    }
+    */
+
+    private final CursorManager cursorManager = mock(CursorManager.class);
+
+    @Before
+    public void setupMock() throws IOException {
+        doCallRealMethod().when(cursorManager).updatePartitions(any(), any());
+        doCallRealMethod().when(cursorManager).fromNewestAvailableOffsets(any(), any());
+        doCallRealMethod().when(cursorManager).fromOldestAvailableOffset(any(), any());
+    }
+
+    private void run(@Nullable String initialOffset, String oldestAvailableOffset, String newestAvailableOffset, @Nullable String expectedOffset) throws IOException {
+        when(cursorManager.getCursors("test")).thenReturn(initialOffset == null ? emptyList() : singletonList(new Cursor("0", initialOffset)));
+        cursorManager.updatePartitions("test", singletonList(new Partition("0", oldestAvailableOffset, newestAvailableOffset)));
+        if (expectedOffset != null) {
+            verify(cursorManager).onSuccess("test", new Cursor("0", expectedOffset));
+        } else {
+            verifyNoMoreInteractions(cursorManager);
+        }
+    }
+
+    @Test
+    public void shouldUpdatePartitionWhenNoCursorAndLastConsumedOffsetNoLongerAvailable() throws IOException {
+        run(null, "10", "20", "10");
```

Comment:
I don't know the purpose of `updatePartitions()` because it's not used, yet. :) But be aware of that if you store it that way it means you have already consumed the offset `"10"`. So if you connect next time with Nakadi offsets:
```
X- Nakadi-Offsets: {"partition": "A", "offset": "10"}
```
the first event you get is the one with offset `"11"`.